velox/exec/ExchangeClient.cpp

```diff
@@ -138,13 +138,19 @@ ExchangeClient::next(uint32_t maxBytes, bool* atEnd, ContinueFuture* future) {
   return pages;
 }
 
+folly::Executor* ExchangeClient::executor() const {
+  if (executor_ != nullptr) {
+    return executor_;
+  }
+  return &folly::QueuedImmediateExecutor::instance();
```

Comment:
Since we believe this is not safe, should we require non-null executor to be specified and document that it cannot be folly::QueuedImmediateExecutor ?